component/main.jsonnet

```
@@ -28,8 +28,205 @@ local net_pol = kube.NetworkPolicy('allow-from-waf-namespaces') {
+  ],
+};
+local rbac_aggregated_gateway_edit = kube.ClusterRole('networking-gatewayapi-aggregated-edit') {
+  metadata: {
+    labels: {
+      'rbac.authorization.k8s.io/aggregate-to-edit': 'true',
+    },
+    name: 'networking-gatewayapi-aggregated-edit',
+  },
+  rules: [
+    rules_crd_gateway,
+    rules_resources_gateway_ronly,
+    rules_resources_gateway_read {
+      verbs+: [ 'create', 'update', 'patch', 'delete' ],
+    },
+  ],
+};
+local rbac_aggregated_gateway_admin = kube.ClusterRole('networking-gatewayapi-aggregated-admin') {
+  metadata: {
+    labels: {
+      'rbac.authorization.k8s.io/aggregate-to-admin': 'true',
+    },
+    name: 'networking-gatewayapi-aggregated-admin',
+  },
+  rules: [
+    rules_crd_gateway,
+    rules_resources_gateway_ronly,
+    rules_resources_gateway_read {
+      verbs+: [ 'create', 'update', 'patch', 'delete' ],
+    },
+  ],
+};
+
+local rules_crd_airlock = {
+  apiGroups: [ 'apiextensions.k8s.io' ],
+  resources: [ 'customresourcedefinitions' ],
+  resourceNames: [
+    'accesscontrolpolicies.microgateway.airlock.com',
+    'accesscontrols.microgateway.airlock.com',
+    'apiprotections.microgateway.airlock.com',
+    'contentsecurities.microgateway.airlock.com',
+    'contentsecuritypolicies.microgateway.airlock.com',
+    'csrfprotections.microgateway.airlock.com',
+    'denyrules.microgateway.airlock.com',
+    'envoyclusters.microgateway.airlock.com',
+    'envoyconfigurations.microgateway.airlock.com',
+    'envoyhttpfilters.microgateway.airlock.com',
+    'gatewayparameters.microgateway.airlock.com',
+    'graphqls.microgateway.airlock.com',
+    'headerrewrites.microgateway.airlock.com',
+    'identitypropagations.microgateway.airlock.com',
+    'jwks.microgateway.airlock.com',
+    'jwts.microgateway.airlock.com',
+    'limits.microgateway.airlock.com',
+    'oidcproviders.microgateway.airlock.com',
+    'oidcrelyingparties.microgateway.airlock.com',
+    'openapis.microgateway.airlock.com',
+    'parsers.microgateway.airlock.com',
+    'redisproviders.microgateway.airlock.com',
+    'sessionhandlings.microgateway.airlock.com',
+    'sidecargateways.microgateway.airlock.com',
+    'telemetries.microgateway.airlock.com',
+    'tokenexchanges.microgateway.airlock.com',
+  ],
+  verbs: [ 'get' ],
+};
+local rules_resources_airlock = {
+  apiGroups: [ 'microgateway.airlock.com' ],
+  resources: [
+    'accesscontrolpolicies',
+    'accesscontrols',
+    'apiprotections',
+    'contentsecurities',
+    'contentsecuritypolicies',
+    'csrfprotections',
+    'denyrules',
+    'envoyclusters',
+    'envoyconfigurations',
+    'envoyhttpfilters',
+    'gatewayparameters',
+    'graphqls',
+    'headerrewrites',
+    'identitypropagations',
+    'jwks',
+    'jwts',
+    'limits',
+    'oidcproviders',
+    'oidcrelyingparties',
+    'openapis',
+    'parsers',
+    'redisproviders',
+    'sessionhandlings',
+    'sidecargateways',
+    'telemetries',
+    'tokenexchanges',
+  ],
```

why the explicit resources? wouldn't it be much better (and more future-proof) to have `resources: ['*']`?

```suggestion
  resources: [ '*' ],
```